python/pyspark/sql/tests/test_utils.py

```diff
@@ -16,18 +16,283 @@
 # limitations under the License.
 #
 
+import unittest
+from prettytable import PrettyTable
```

Comment:
Is this to check the equality? I think you can just sort and compare the rows.

Reply:
the prettytable is to help with formatting the error message when there are row inequalities -- it helps display all the rows that differ

Reply:
Can you see if you can leverage builtin module such as https://docs.python.org/3/library/difflib.html? I would like to avoid adding a dependency unless if we really need. How's the output if we use this library?

Reply:
@HyukjinKwon I don't think difflib will work unfortunately. difflib is to compare strings. I can convert PySpark df -> pandas df -> str, and put that into difflib, but then the output isn't clear because it only shows the exact characters that are different. 

prettytable is nice because it can stack the rows and color-code them. I remember in a design discussion we said [prettytable](https://github.com/jazzband/prettytable) may be okay to add as a dependency, since it is popular (8.2million downloads per month). Another option is reimplementing similar functionality to prettytable, but I think it might make sense to just use what already exists?

For example, here's the difference in output for difflib and prettytable for a simple pyspark df

`df1 = self.spark.createDataFrame(
            data=[
                ("1", 1000.00),
                ("2", 3000.00),
            ],
            schema=["id", "amount"],
        )
        df2 = self.spark.createDataFrame(
            data=[
                ("1", 1001.00),
                ("2", 3000.00),
            ],
            schema=["id", "amount"],
        )`

difflib:
<img width="173" alt="Screenshot 2023-07-05 at 9 17 11 AM" src="https://github.com/apache/spark/assets/68875504/444f6091-57a8-4d1d-a323-0a5b5b0a9c82">

prettytable:
<img width="494" alt="Screenshot 2023-07-05 at 9 19 44 AM" src="https://github.com/apache/spark/assets/68875504/5a6f4123-fcff-4405-ada0-210ac8e0cb9a">